src/main-oe/sgxlkl_params.c

```diff
@@ -38,285 +43,107 @@ struct sgxlkl_config_elem
     };
 };
 
+typedef struct json_callback_data
+{
+} json_callback_data_t;
+
+#define DEFAULT_SGXLKL_VERBOSE 0
+#define DEFAULT_SGXLKL_CWD "/"
+#define DEFAULT_SGXLKL_GW4 "10.0.1.254"
+/* The default heap size will only be used if no heap size is specified and
+ * either we are in simulation mode, or we are in HW mode and a key is provided
+ * via SGXLKL_KEY.
+ */
+#define DEFAULT_SGXLKL_OE_HEAP_PAGE_COUNT 8192 /* 8192 * 4K = 32MB */
+#define DEFAULT_SGXLKL_HEAP_SIZE 200 * 1024 * 1024
+#define DEFAULT_SGXLKL_HOSTNAME "lkl"
+#define DEFAULT_SGXLKL_IP4 "10.0.1.1"
+#define DEFAULT_SGXLKL_MASK4 "24"
+#define DEFAULT_SGXLKL_MAX_USER_THREADS 256
+#define DEFAULT_SGXLKL_ESLEEP 16000
+#define DEFAULT_SGXLKL_ETHREADS 1
+#define DEFAULT_SGXLKL_ESPINS 500
+#define DEFAULT_SGXLKL_STACK_SIZE 512 * 1024
+#define DEFAULT_SGXLKL_SWIOTLB 1
+#define DEFAULT_SGXLKL_TAP "sgxlkl_tap0"
+#define DEFAULT_SGXLKL_WG_IP "10.0.2.1"
+#define DEFAULT_SGXLKL_WG_PORT 56002
+#define DEFAULT_SGXLKL_KERNEL_CMD "mem=32M"
+#define DEFAULT_SGXLKL_HOSTNET false
+#define DEFAULT_SGXLKL_TAP_MTU 0
+
+// clang-format off
 static struct sgxlkl_config_elem sgxlkl_config[] = {
```

Comment:
I ported it to the libjson, but there is no schema-based host config, see other comments about host config. We can add that at some point, but it's not necessary now.

Reply:
Isn't this what we discussed offline? I thought the enclave and host config should be handled in the same way, both using schemas?

Reply:
Just to reiterate: whenever someone uses the term "host config" it is not clear what they are referring to. E.g. @prp includes the host-compiled enclave config in the host config sometimes (but not always). 

The enclave configuration (enclave-config.json, `sgxlkl_enclave_config_t`) is handled identically on the host and in the enclave, they even use the same parser now (after I simplified everything down, the difference was really only the string-encoded numbers + some parser options). 

There is an empty schema for a host-config that someone set up. In theory, we could fill that in and auto-generate a struct for it too. The rough code structure for this is in place in the launcher (`sgxlkl_host_config_t`), but it would require a significant amount of refactoring in `sgxlkl_run_oe.c`, which will likely be replaced with a re-implementation in C++, so I think it's a waste of time to do that now.

Reply:
No, I don't :-) The 'host config' is a user-facing JSON file and only includes parameters that the launcher cares about and that don't have security implications.

Reply:
@wintersteiger I think it would be good if you edit the PR description to say what's included here and what will be done in future PRs and why.

Reply:
Yeah, that description is definitely not right anymore. What is or will be included isn't decided yet. E.g., if you want me to add a host config schema, I can do that. But, that can also be a separate PR (if so, ideally by someone else as I'm running way over budget here). 

Aside: @prp yes you did, e.g. last Teams call! No worries though, it's good that we have a definition now :-)